src/Functions/IFunctionCustomWeek.h

```diff
@@ -56,6 +56,21 @@ class IFunctionCustomWeek : public IFunction
                 : is_not_monotonic;
         }
 
+        if (checkAndGetDataType<DataTypeDateTime64>(&type))
+        {
+
+            const auto & left_date_time = left.safeGet<DateTime64>();
+            TransformDateTime64<Transform> transformer_left(left_date_time.getScale());
```

Comment:
As far as I understand, you need to use Transform::FactorTransform in this place, or am I wrong?

Reply:
oh, yes, exactly... fixed